modules/afmongodb/afmongodb-worker.c

```diff
@@ -146,13 +199,9 @@ _worker_connect(LogThreadedDestWorker *s)
       read_prefs = mongoc_collection_get_read_prefs(self->coll_obj);
     }
 
-
   if (!_check_server_status(self, read_prefs))
     {
-      mongoc_collection_destroy(self->coll_obj);
-      self->coll_obj = NULL;
-      mongoc_client_pool_push(owner->pool, self->client);
-      self->client = NULL;
+      _worker_disconnect(s);
```

Comment:
This commit has a few refactor steps as well, like this one.
Please consider creating a separate refactor commit for them next time. :)

Reply:
usually, I do, but  _worker_disconnec had changed too, therefore I did not want to add a version that reflects the changes duplicated again here than replace the whole block separately